### PR TITLE
#162082169 Setup testing configuration

### PR DESCRIPTION
**Pull request by queenfiona**
#### What does this PR do?
Create a branch for setting up testing configuration
#### Description of Task to be completed?
Setup testing configuration
#### How should this be manually tested?
Clone this repository, cd into it, in terminal type:
 export FLASK_ENV='testing'
python -m unittest discover
#### What are the relevant pivotal tracker stories?
#162082169

### CODE REVIEW

File: run.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 """docstring for run.py app import create_app."""
 from app import create_app
 
-app = create_app()
+app = create_app('development')
```
Comment (borenho) [inline run.py]: Don't hard code the config here. just pass `config` then figure out what config it is in the lines that follow. To allow for dynamic app creation

**Review by kaka-ruto — CHANGES_REQUESTED** [review]

File: run.py (hunk 1 above)
Comment (queenfiona) [reply run.py]: I have made the flask environment dynamic in commit https://github.com/queenfiona/SendITc3/pull/2/commits/c21923e331af6d27c2ab20847b1e83c2278bd3ca

File: requirements.txt [hunk 2]
```diff
@@ -22,9 +23,11 @@ psycopg2-binary==2.7.5
 PyJWT==1.6.4
 pylint==2.1.1
 pyparsing==2.3.0
+python-dotenv==0.9.1
 pytz==2018.7
 requests==2.20.0
 six==1.11.0
+SQLAlchemy==1.2.14
```
[the remark this answers is not in the document]
Comment (queenfiona) [reply requirements.txt]: Removed SQLAlchemy and flask-SQLAlchemy in commit https://github.com/queenfiona/SendITc3/pull/2/commits/b5d108dbd48e23f0ead81fb9dbd487450686dca7

File: requirements.txt [hunk 3]
```diff
@@ -7,6 +7,7 @@ coverage==4.5.1
 Flask==1.0.2
 Flask-JWT-Extended==3.13.1
 Flask-RESTful==0.3.6
+Flask-SQLAlchemy==2.3.2
```
[the remark this answers is not in the document]
Comment (queenfiona) [reply requirements.txt]: Removed Flask-SQLAlchemy  in commit https://github.com/queenfiona/SendITc3/pull/2/commits/b5d108dbd48e23f0ead81fb9dbd487450686dca7

File: app/__init__.py [hunk 4]
```diff
@@ -1,11 +1,15 @@
...
 
-def create_app():
+def create_app(config):
     """Docstring for create_app method."""
-    app = Flask(__name__)
+    app = Flask(__name__, instance_relative_config=True)
+    app.config.from_object(app_config[config])
+    app.register_blueprint(version_2)
     destroy_tables()
     create_tables()
```
[the remark this answers is not in the document]
Comment (queenfiona) [reply app/__init__.py]: Removed the  destroy_tables method in commit https://github.com/queenfiona/SendITc3/pull/2/commits/bab88db356d52efdb1e426e90b48eb036c329495

File: app/db_config.py [hunk 5]
```diff
@@ -1,7 +1,10 @@
 """Docstring for db_config file."""
 import psycopg2
+# import os
+from instance.config import app_config
 
-url = "host = localhost dbname = sendit user = app password = app"
+# env = os.getenv("FLASK_ENV")
+url = app_config['testing'].DATABASE_URL
```
[the remark this answers is not in the document]
Comment (queenfiona) [reply app/db_config.py]: Made flask environment variables dynamic in commit https://github.com/queenfiona/SendITc3/pull/2/commits/c21923e331af6d27c2ab20847b1e83c2278bd3ca

File: instance/config.py [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+class ProductionConfig(Config):
+    """docstring for ProductionConfig."""
+
+    pass
+
+
+app_config = {
+    "development": DevelopmentConfig,
+    "testing": TestingConfig,
+    "default": DevelopmentConfig
```
[the remark this answers is not in the document]
Comment (queenfiona) [reply instance/config.py]: Added "production": ProductionConfig in app_config in commit https://github.com/queenfiona/SendITc3/pull/2/commits/a228734ea7592cf2787091542fa1aac4c1464e19

File: instance/config.py [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+    """docstring for StagingConfig."""
+
+    DEBUG = True
+    TESTING = True
+
+
+class ProductionConfig(Config):
+    """docstring for ProductionConfig."""
+
+    pass
```
[the remark this answers is not in the document]
Comment (queenfiona) [reply instance/config.py]: Added production config variables in commit https://github.com/queenfiona/SendITc3/pull/2/commits/a228734ea7592cf2787091542fa1aac4c1464e19

File: app/db_config.py [hunk 8]
```diff
@@ -51,8 +54,8 @@ def create_tables():
         try:
             cursor.execute(table)
             connection_to_db.commit()
-        except Exception as e:
-            raise e
+        except:
+            return {"message": "could not create tables"}
```
[the remark this answers is not in the document]
Comment (queenfiona) [reply app/db_config.py]: Removed try/except from create_tables method in db_config file in commit https://github.com/queenfiona/SendITc3/pull/2/commits/32a07550c16da98aefee0a91952cfa5f3458d4a0

File: .env [hunk 9]
```diff
@@ -0,0 +1,5 @@
+export FLASK_ENV=testing
+export FLASK_DEBUG=True
+export FLASK_APP=run.py
+export DATABASE_URL="host='localhost' port='5432' dbname='sendit' user='app' password='app'"
+export DATABASE_TESTING_URL="host='localhost' port='5432' dbname='sendit_test' user='postgres' password='postgres'"
```
[the remark this answers is not in the document]
Comment (queenfiona) [reply .env]: Untracked .env file in commit https://github.com/queenfiona/SendITc3/pull/2/commits/74eea23678a81332d8c0b6096d484c09e2b1270c

File: .env [hunk 10]
```diff
@@ -0,0 +1,5 @@
+export FLASK_ENV=testing
```
[the remark this answers is not in the document]
Comment (queenfiona) [reply .env]: Added .env in .ignore file, it can no longer be tracked.

[the remark this answers is not in the document]
Comment (queenfiona) [reply .env]: env

**Review by kaka-ruto — CHANGES_REQUESTED** [review]

**Review by kaka-ruto — APPROVED** [review]

**Pull request merged**